### ISSUE TITLE
readMSData2 for level 1:...

### ISSUE THREAD

**Issue by lgatto** [lgatto/MSnbase#94]
support something like `msLevel = 1:2`.

Code would be something like:

``` r
spidx <- which(fullhd$msLevel %in% msLevel)
## iterate over spidx
```

**Comment by lgatto** [comment]
- what about methods that process spectra, or should only process certain levels?
- what about plotting?
- have an `msLevel` argument, to control which to process?
- define params defining how to process different levels?

**Comment by lgatto** [comment]
@jotsetung - here's how I see the development of `readMSData` and the resulting `MSnExp` and `OnDiskMSnExp`, and I curious to hear your thoughts.

- Settle for `OnDiskMSnExp` as the main data structure, and this is what `readMSData` would return.
- `OnDiskMSnExp` instances are just called `MSnExp` instances that live on disk (see the new `show` method, that calls object `MSnExp` and specifies `on disk` or `in memory`). No need to explicitly call the instances as `MSnExp` and `OnDiskMSnExp` as (1) it is confusing and (2) not necessary as their behaviours should be identical.
- If, at any point, the user wants on in memory instance, they convert it with `xmem <- inMemory(x)`. 
- `readMSData` creates and `OnDiskMSnExp` instance with all spectra (this is fast anyway). - If `msLevel` is specified by the user (default would be `missing`), then, before returning the object, it is subset accordingly. This simplifies the code substantially. 

There is one important effect of having a mix of MS levels in an `MSnExp` (whether on disk on in memory). The spectrum processing methods will have to be made aware of this. I am not sure yet what the best approach is for now, but that does not invalidate the above, as it is always possible and fast to subset an `MSnExp` object to keep only the desired MS levels. The latter might even be wrapped into a function to save some typing - something like 

```r
x1 <- filterMsLevel(x, 1)
x2 <- filterMsLevel(x, 2)
```
instead of 
```r
x1 <- x[msLevel(x) == 1]
x3 <- x[msLevel(x) == 2]
```

**Comment by jotsetung** [comment]
Just having one object instead of two would indeed simplify most, if not all, methods; specifically, there will be no need to have all methods implemented twice.
The `MSnExp` will however need a slot `backend` to specify whether the data is expected to be in memory, on disk, or eventually in future in SQLite file(s).
The rest should then be really simple to implement; only the main spectrum accessor methods need to be aware of this slot and, depending on its value, read the data from disk.

**Comment by lgatto** [comment]
Let's keep the `MSnExp` for now (see this [comment](https://github.com/lgatto/MSnbase/issues/101#issuecomment-229469276)].

**Comment by lgatto** [comment]
@jotsetung - I would like to revert `readMSData` to the old version (i.e. in memory) and concentrate on `readMSData2` (will be renamed later, when we are ready for official release), that reads all MS levels. Are you OK with that?

**Comment by jotsetung** [comment]
Sure. I'll then fix and update the unit tests for `OnDiskMSnExp`.

**Comment by lgatto** [comment]
Reopening this issue to update parameters:

- [X] in comment b5ed2a0e2e487bab7381c10379532e71d93bad4e, I have removed the `removePeaks` and `clean` parameters.
- [ ] in that same commit, I have also set `smoothed` and `centroided` defaults to `NA`
- but, I am also wondering of `centroided` and `smoothed` shouldn't be upgraded so that is becomes possible to defined them to different MS levels. In a typical Thermo files, MS1 levels would be in profile mode, while MS2 spectra would be centroided.
